addons/mrp/tests/test_byproduct.py

```diff
@@ -529,3 +530,38 @@ def test_3_steps_byproduct(self):
         self.assertEqual(len(postprod_picking.move_ids), 2)
         self.assertEqual(postprod_picking.move_ids.product_id, final_product + byproduct)
         self.assertEqual(postprod_picking.location_dest_id, self.warehouse.lot_stock_id)
+
+    def test_over_produce_by_products_with_cost_share(self):
+        """
+        Test that overproducing by porducts with a set cost share is
+        well behaved (as it should rely on the merge move) for the
+        extra move.
+        """
+        # Create new MO
+        self.env.user.groups_id = [Command.link(self.ref('mrp.group_mrp_byproducts'))]
+        self.bom_byproduct.byproduct_ids.cost_share = 3.3
+        workcenter = self.env['mrp.workcenter'].create({
+            'name': 'Nuclear Workcenter',
+        })
+        self.bom_byproduct.operation_ids = [Command.create({
+            'name': "Create by products",
+            'workcenter_id': workcenter.id,
+        })]
+        self.bom_byproduct.byproduct_ids.operation_id = self.bom_byproduct.operation_ids
+
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.product_id = self.product_a
+        mo_form.product_qty = 1.0
+        mo = mo_form.save()
```

Comment:
```suggestion
        mo = self.env['mrp.production'].create({
            'product_id': self.product_a.id,
            'product_qty': 1.0,
        })
```
No need to use a `Form` if you don't need any interaction from the form itself. It's a bit faster to go through the `create()` directly.